client/finality-grandpa/src/authorities.rs

```diff
@@ -714,6 +714,17 @@ impl<N: Ord + Clone> AuthoritySetChanges<N> {
 		}
 	}
 
+	pub(crate) fn insert(&mut self, block_number: N) {
```

Comment:
Authority set changes are never pruned IIUC. They are needed to provide warp proofs. When ancient blocks are downloaded after the warp sync we want to collect authority set changes, so that this node could provide warp sync service itself.

Reply:
Yes, I did not remember correctly, this pruning is only true for babe. Indeed it would be an issue for warp proof anyway.